modules/ding_variables/ding_variables.module

```diff
@@ -0,0 +1,85 @@
+<?php
+/**
+ * @file
+ * Define modules hooks.
+ */
+
+/**
+ * Implements hook_menu().
+ */
+function ding_variables_menu() {
+  $items = array();
+
+  $items['ding/variables'] = array(
+    'title' => 'Display variables as JSON output',
+    'access callback' => TRUE,
+    'delivery callback' => 'drupal_json_output',
+    'page callback' => 'ding_variables_display',
+  );
+
+  return $items;
+}
+
+/**
+ * Provide variables to the json output.
+ *
+ * @return array
+ *   List of all variables to the output.
+ */
+function ding_variables_display() {
+  // TODO: Find a way to remove hardcode.
+  $carousels = array(
+    'carousel',
+  );
+  $carousel_configs = array();
+  foreach ($carousels as $carousel) {
+    $carousel_configs[] = _ding_variables_get_carousel_configs($carousel);
+  }
+
+  $variables = array(
+    'ting_search_profile',
+    'ting_search_results_per_page',
+    'ting_search_result_message_limit',
+    'ting_search_url',
+    'ting_sort_default',
+    'ting_ranking_custom',
+    'ting_ranking_fields',
+    'ting_boost_fields',
+    'ting_agency',
+  );
```

Comment:
Please note that within the 4.3.x the SAL release these variables have been moved into the opensearch module and I think most of them simply can be renamed to opensearch_*.

@kasperg can all these simply be renamed?

Reply:
Yes. Check [`opensearch_migrate_settings()`](https://github.com/ding2/ding2/blob/master/modules/opensearch/opensearch.install#L18) for new names.

One notable exception is `ting_agency` - [that is bound to change later on though](https://platform.dandigbib.org/issues/3322).